import "ConnectionMessage.proto";
import "QueryChunk.proto";
import "ClientActions.proto";
import "PingTime.proto";

message ClientMessage {
  enum PacketContent
  {
    CONNECTION = 0;
    IDENTIFICATION = 1;
    QUERYCHUNK = 2;
    ACTION = 3;
    GETPLAYER = 4;
    PING = 5;
    CHAT = 6;
  }
  required PacketContent content = 1;
  optional ConnectionMessage co = 2;
  optional QueryChunk queryChunk = 3;
  optional ClientActions actions = 4;
  optional PingTime ping = 5;
  optional string chat = 6;
}
